move config values into variables to easily change it

tecbird opened the pull request:
<!--

Before working on a contribution, you must determine on which branch you need to work:
- Bug fix: choose the latest maintenance branch `10.5`
- Feature/Improvement: choose `11.x` 

> All bug fixes merged into the latest maintenance branch are also merged to the current dev branch (`11.x`) on a regular basis.

## Please make sure your PR complies with all of the following points: 
- [ ] Read and accept our [contributing guidelines](/CONTRIBUTING.md) before you submit a PR.
- [ ] Features need to be proper documented in `doc/` 
- [ ] Bugfixes need a short guide how to reproduce them -> target branch is the oldest supported maintenance branch, e.g. `10.5` (see Readme.md for the list of supported versions)
- [ ] Meet all coding standards (see PhpStan actions) 

**Don't submit a PR if it doesn't comply, it'll be closed without a comment!**
-->  
  

## Changes in this pull request  
Resolves #
move config values into variables to easily change it

## Additional info

tecbird committed:
move config values into variables to easily change it

robertSt7 commented:
@tecbird Could you please define maxRenderItems in https://github.com/pimcore/pimcore/blob/11.x/bundles/AdminBundle/public/js/pimcore/object/tags/abstractRelations.js, replace the maximum render items (10) in every realtion type with maxRenderItems and pass it as parameter with the default value 10, where necassary. Thanks a lot

aryaantony92 committed:
Fix xliff import

dvesh3 committed:
[Translations]: Fix WYSIWYG translation not editable - resolves #13761

frithjof committed:
replace lrm characters with - for validkeys

passioneight committed:
fix: twig sandbox error for "default" filter

resolves #13788

dvelguru committed:
Typo (#13805)

Typo

dvesh3 committed:
Bump symfony/contracts to 3.2 & knplabs/knp-paginator-bundle to 6.0.0 (#13766)

* Bump symonfy/contracts to ^3.2 & knplabs/knp-paginator-bundle to ^6.0.0

* bump knplabs/knp-paginator-bundle to ^6.0.0 - fix service definitions

mattamon committed:
Apply php-cs-fixer changes

robertSt7 committed:
Fix: User type (#13810)

* Fix: User type

* Fix: o_ prefix

torqdev committed:
make custom reports scrollable when needed (#13623)

Co-authored-by: Cameron Jenkins <cameron.jenkins@torqit.ca>

blankse committed:
[Bug]: After moving data object the query tables are still filled with inherited data from source location (#13735)

mcop1 committed:
[Task] Fix db column quoting (#13816)

mattamon committed:
Fix: Do not cast resources to string

Commit:
prevent deprecated error "Passing null to parameter #1 ($string) of type string is deprecated" (#13828)

SamyMP committed:
Let user select an uneditable field in grid view (#13615)

Replaced the ExtJS MessageBox component popup with a toast to prevent blocking the user from selecting a field

mcop1 committed:
[Task]: Fix possible errors after applying php native types  (#13821)

* [Task] Fix db column quoting

* [Task]: Fix possible errors after applying php native types

* Revert "[Task] Fix db column quoting"

This reverts commit 8e97b73df71a5bf7668d100815de78eee92e383f.

* [Task]: Fix possible errors after applying php native types

dvesh3 committed:
Do not cast resources to string - follow up to #13834

robertSt7 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

dvesh3 committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	models/DataObject/ClassDefinition/Data/Select.php

dvesh3 committed:
Fix UserTest extends wrong TestCase

dvesh3 committed:
Update POEditor translations

blankse committed:
Translation::getByKeyLocalized(): Remove unnecessary func_get_args()

Corepex committed:
[Task] Substitute `method exists` with interface check in field definitions (#13846)

* added ClassSavedInterface to several classes, removed method_exists

* Update AdvancedManyToManyRelation.php

added $params parameter

* Update models/DataObject/ClassDefinition/Data/Fieldcollections.php

added BC layer

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* Update Fieldcollections.php

added comment to remove method_exists

* added BC layer

* Update models/DataObject/ClassDefinition/Data/Objectbricks.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* Update models/DataObject/ClassDefinition/Data/Localizedfields.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* passed parameters to classSaved function

* removed interface from classes

* removed interface from classes

* Update models/DataObject/ClassDefinition/Data/Fieldcollections.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* added upgrade notes

* changed interface

* added type

* updated upgrade notes

* added unknown parameter to ignore list

* added unknown parameter to ignore list

* fixed phpstan formatting

* fixed phpstan formatting

* changed condition order

Co-authored-by: Sebastian Blank <blank@data-factory.net>

robertSt7 committed:
Apply php-cs-fixer changes

blankse committed:
Site: getBy* methods shouldn't return string

mattamon committed:
Fix: Do not cast resources to string

brusch committed:
[Translations] Cleaned up abandoned files

brusch committed:
Update zh_Hans.json (POEditor.com)

brusch committed:
Update cs.json (POEditor.com)

brusch committed:
Update nl.json (POEditor.com)

brusch committed:
Update fr.json (POEditor.com)

brusch committed:
Update de.json (POEditor.com)

brusch committed:
Update hu.json (POEditor.com)

brusch committed:
Update it.json (POEditor.com)

brusch committed:
Update ja.json (POEditor.com)

brusch committed:
Update fa.json (POEditor.com)

brusch committed:
Update pl.json (POEditor.com)

brusch committed:
Update pt.json (POEditor.com)

brusch committed:
Update pt_BR.json (POEditor.com)

brusch committed:
Update ru.json (POEditor.com)

brusch committed:
Update sk.json (POEditor.com)

brusch committed:
Update es.json (POEditor.com)

brusch committed:
Update sv.json (POEditor.com)

brusch committed:
Update sv_FI.json (POEditor.com)

brusch committed:
Update th.json (POEditor.com)

brusch committed:
Update tr.json (POEditor.com)

brusch committed:
Update uk.json (POEditor.com)

brusch committed:
Update cs.extended.json (POEditor.com)

brusch committed:
Update nl.extended.json (POEditor.com)

brusch committed:
Update fr.extended.json (POEditor.com)

brusch committed:
Update de.extended.json (POEditor.com)

brusch committed:
Update hu.extended.json (POEditor.com)

brusch committed:
Update it.extended.json (POEditor.com)

brusch committed:
Update pl.extended.json (POEditor.com)

brusch committed:
Update sk.extended.json (POEditor.com)

brusch committed:
Update es.extended.json (POEditor.com)

brusch committed:
Update th.extended.json (POEditor.com)

brusch committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

brusch committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	bundles/CoreBundle/translations/pt-br.json
#	bundles/CoreBundle/translations/pt_BR.json
#	bundles/CoreBundle/translations/sv-fi.json
#	bundles/CoreBundle/translations/sv_FI.json
#	bundles/CoreBundle/translations/zh-CN.json
#	bundles/CoreBundle/translations/zh_Hans.json
#	models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php
#	models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php
#	models/DataObject/ClassDefinition/Data/Classificationstore.php
#	phpstan-baseline.neon

dvesh3 committed:
[Web2Print] Render twig templates in a sandbox (#13519)

* [Web2Print] Render twig templates in a sandbox - follow up to #13347

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

Co-authored-by: Sebastian Blank <blank@data-factory.net>

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>
Co-authored-by: Sebastian Blank <blank@data-factory.net>

dvesh3 committed:
Apply php-cs-fixer changes

robertSt7 committed:
Fix: multiple naming (#13855)

dvesh3 committed:
Apply php-cs-fixer changes

Corepex committed:
pass $params to classSaved method in postSave trait

Corepex committed:
Revert "pass $params to classSaved method in postSave trait"

This reverts commit 6a9879ce4db518a44c73d7589a92a5a91fc4cdf8.

jheimbach committed:
Logout from custom firewall always deletes complete session (#13825)

* register logoutlistener only on pimcore_admin firewall

* clear only admin session

* set response to logout event object

Corepex committed:
[Task] Enabled $param parameter for classSaved method in `classSavedInterface` (#13857)

* enabled $param parameter for classSaved method

* removed ignored error

* pass $params to classSaved method in postSave trait

* removed deprecated code

* added upgrade notes

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

jdreesen committed:
Fix `property_exists(): Passing null to parameter #2 ($property) of type string is deprecated` (#13854)

* Fix `property_exists(): Passing null to parameter #2 ($property) of type string is deprecated`

* Adapt @param annotation

limenet committed:
HeadlessChrome: modify config via event listener (#13806)

* merge global options with options passed via event

* use modifyConfig event to modify HeadlessChrome options

dvesh3 committed:
Fix adding existing translation results in error in Pimcore admin (#13866)

* Fix adding existing translation results in error in Pimcore admin - #13860

* Fix adding existing translation results in error in Pimcore admin - #13860

* Fix adding existing predefined asset metadata - #13860

mattamon committed:
Apply php-cs-fixer changes

blankse committed:
Fix max height of grouped brick menu (#13672)

* Fix max height of grouped brick menu

* Review

* Review

mcop1 committed:
[Bug]: TestDataHelper doesn´t respect localizedFields in assertIsEqua… (#13872)

* [Bug]: TestDataHelper doesn´t respect localizedFields in assertIsEqual method

* [Bug]: TestDataHelper doesn´t respect localizedFields in assertIsEqual method
Removed newlines in services.yaml

* [Bug]: TestDataHelper doesn´t respect localizedFields in assertIsEqual method
Removed newlines in services.yaml

* Apply suggestions from code review

Co-authored-by: Sebastian Blank <blank@data-factory.net>

Co-authored-by: Sebastian Blank <blank@data-factory.net>

fashxp committed:
Apply php-cs-fixer changes

Corepex committed:
[Bug] Fixed string comparison issue to detect changed while saving localized fields (#13853)

* made string compare strict

* adapted === as new default, added simple compare for select

* added upgrade notes

* removed upgrade notes

* [Bug]: Check create, update and new mutation for undefined array keys

Co-authored-by: Marco Perberschlager <marco.perberschlager@pimcore.com>

mcop1 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

# Conflicts:
#	doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

blankse committed:
PhpStan: Remove ignored error pattern from baseline file (#13870)

* PhpStan: Remove ignored error pattern from baseline file

* Fix PhpStan error

fashxp committed:
Further type fixes (#13856)

* further type fixes

* Removed ignored error from phpstan-baseline.neon

* Revert "Removed ignored error from phpstan-baseline.neon"

This reverts commit 7216b5e04b748c4916ffe0ee40f58f0a4224c7be.

Co-authored-by: Marco Perberschlager <marco.perberschlager@pimcore.com>

dvesh3 committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	models/DataObject/ClassDefinition/Data/BooleanSelect.php
#	models/DataObject/Traits/ObjectVarTrait.php
#	tests/Support/Helper/DataType/TestDataHelper.php

dvesh3 committed:
Merge remote-tracking branch 'origin/11.x' into 11.x

dvesh3 committed:
Apply php-cs-fixer changes

fashxp committed:
fixed #13871 - [Bug][Pimcore 11] [Assets] Grid metadata batch edit all does not work

PMD-TorqIT committed:
Checkbox fix.

Corepex committed:
[Improvement] Revalidate phpdoc types (#13755)

* changed doctypes

* changed doctypes

* changed doctypes

* changed doctypes

* changed doctypes

* Update bundles/EcommerceFrameworkBundle/CartManager/AbstractCart.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* Update bundles/EcommerceFrameworkBundle/EnvironmentInterface.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* fixed phpstan

* fixed phpstan

* removed phpstan error

* changed doctypes according to new type

* changed doctypes according to new type

* changed doctypes according to new type

* revert changes

* changed doctypes according to new type

* changed doctypes according to new type

* changed doctypes according to new type

* changed doctypes according to new type

* fixed phpstan issues

* Update CustomLayout.php

* Update CustomLayout.php

* Update models/DataObject/ClassDefinition/CustomLayout.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

Co-authored-by: Sebastian Blank <blank@data-factory.net>
Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

mcop1 committed:
Apply php-cs-fixer changes

brusch committed:
[Tests] docker compose should use debug image as stated in the docs

brusch committed:
[Routing] Do not invoke dynamic routing for non-default context requests

dpfaffenbauer committed:
Folder Preview stream_get_contents(): Argument #1 ($stream) must be of type resource, null given

Folder Preview $tileThumb->getStream can return null

dpfaffenbauer committed:
Fix fclose(): supplied resource is not a valid stream resource

This happens quite often on our setup, not sure why, but this fixes the issue.

blankse committed:
Fix: Definition::isWritable(): Return value must be of type bool, string returned (#13884)

blankse committed:
Unit::setBaseunit() must allow string and null (#13882)

mcop1 committed:
[Improvement]: \Pimcore\Bundle\AdminBundle\GDPR\DataProvider\SentMail should be @internal (#13836)

brusch committed:
Apply php-cs-fixer changes

blankse committed:
User::roles should be a array of intergers (#13850)

blankse committed:
Simplify DirtyIndicatorTrait (#13744)

jremmurd committed:
Fix preGetData missing language param for Data\Wysiwyg (#13837)

* Fix preGetData missing language in Data\Wysiwyg

* Update Wysiwyg.php

blankse committed:
Use Required attributes instead of annotations (#13885)

* Use Required attribute instead of annotations

* Add return types

mcop1 committed:
[Improvement]: Make getVersionDependentDatabaseColumnName case insensitive (#13822)

* [Improvement]: Make getVersionDependentDatabaseColumnName case insensitive

* Update models/DataObject/Service.php

Co-authored-by: Christian F. <christian.feldkirchner@pimcore.com>

* [Improvement]: Make getVersionDependentDatabaseColumnName case insensitive
Make sure to keep the case of the $fieldName parameter

Co-authored-by: Christian F. <christian.feldkirchner@pimcore.com>

mcop1 committed:
Apply php-cs-fixer changes

fashxp committed:
Type fixes (#13877)

* further type fixes

* some type fixes

dvesh3 committed:
Improve version checks with Version::MAJOR_VERSION constant (#13896)

robertSt7 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

brusch committed:
Apply php-cs-fixer changes

blankse committed:
User::twoFactorAuthentication attribute should be null by default (#13848)

* User::twoFactorAuthentication attribute should be null by default
* Check against null
* Remove User::roles types

kingjia90 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	bundles/EcommerceFrameworkBundle/Tracking/Transaction.php
#	bundles/EcommerceFrameworkBundle/src/CartManager/AbstractCart.php
#	bundles/EcommerceFrameworkBundle/src/CartManager/AbstractCartCheckoutData.php
#	bundles/EcommerceFrameworkBundle/src/CartManager/SessionCartCheckoutData.php
#	bundles/EcommerceFrameworkBundle/src/CheckoutManager/CheckoutManagerFactory.php
#	bundles/EcommerceFrameworkBundle/src/CoreExtensions/ClassDefinition/IndexFieldSelection.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Config/DefaultFindologic.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/ProductList/DefaultFindologic.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/ProductList/DefaultMysql.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/ProductList/ElasticSearch/AbstractElasticSearch.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/ProductList/ProductListInterface.php
#	bundles/EcommerceFrameworkBundle/src/Model/AbstractVoucherTokenType.php
#	bundles/EcommerceFrameworkBundle/src/OfferTool/AbstractOfferToolProduct.php
#	bundles/EcommerceFrameworkBundle/src/PaymentManager/V7/Payment/PaymentInterface.php
#	bundles/EcommerceFrameworkBundle/src/Tracking/ProductAction.php
#	bundles/EcommerceFrameworkBundle/src/VoucherService/Reservation.php
#	bundles/EcommerceFrameworkBundle/src/VoucherService/Token.php
#	bundles/EcommerceFrameworkBundle/src/VoucherService/TokenManager/Pattern.php
#	lib/Event/Targeting/BuildConditionEvent.php
#	lib/Image/Adapter.php
#	lib/Mail.php
#	lib/Model/AbstractModel.php
#	lib/Model/ModelInterface.php
#	lib/Tool/Text.php
#	lib/Translation/ExportService/Exporter/Xliff12Exporter.php
#	lib/Twig/Extension/Templating/Placeholder/Container.php
#	lib/Version.php
#	models/Asset.php
#	models/Asset/Image.php
#	models/Asset/Image/Thumbnail/Config.php
#	models/DataObject/AbstractObject.php
#	models/DataObject/ClassDefinition/CustomLayout.php
#	models/DataObject/ClassDefinition/Data/Block.php
#	models/DataObject/ClassDefinition/Data/Classificationstore.php
#	models/DataObject/ClassDefinition/Data/Multiselect.php
#	models/DataObject/ClassDefinition/Data/Objectbricks.php
#	models/DataObject/ClassDefinition/Data/QuantityValue.php
#	models/DataObject/ClassDefinition/Data/UrlSlug.php
#	models/DataObject/ClassDefinition/Data/User.php
#	models/DataObject/ClassDefinition/Layout.php
#	models/DataObject/Data/ElementMetadata.php
#	models/DataObject/Data/ExternalImage.php
#	models/DataObject/Data/GeoCoordinates.php
#	models/DataObject/Data/ImageGallery.php
#	models/DataObject/Data/InputQuantityValue.php
#	models/DataObject/Data/Link.php
#	models/DataObject/Data/ObjectMetadata.php
#	models/DataObject/Data/QuantityValue.php
#	models/DataObject/Data/Video.php
#	models/DataObject/QuantityValue/Unit.php
#	models/DataObject/Traits/FieldcollectionObjectbrickDefinitionTrait.php
#	models/DataObject/Traits/OwnerAwareFieldTrait.php
#	models/Document.php
#	models/Document/Editable/Embed.php
#	models/Document/Editable/Image.php
#	models/Document/Editable/Numeric.php
#	models/Document/Editable/Renderlet.php
#	models/Document/Editable/Select.php
#	models/Document/Editable/Snippet.php
#	models/Document/Editable/Textarea.php
#	models/Document/Editable/Video.php
#	models/Document/Editable/Wysiwyg.php
#	models/Element/Service.php
#	models/Metadata/Predefined.php
#	models/Property/Predefined.php
#	models/Redirect.php
#	models/Schedule/Task.php
#	models/Site.php
#	models/Staticroute.php
#	models/Tool/CustomReport/Config.php
#	models/Tool/Email/Log.php
#	models/Tool/Targeting/Rule.php
#	models/Tool/Targeting/TargetGroup.php
#	models/Translation.php
#	phpstan-baseline.neon

dvesh3 committed:
Merge remote-tracking branch 'origin/11.x' into 11.x

dvesh3 committed:
Fix merge conflicts in https://github.com/pimcore/pimcore/commit/504c00257e8ca9e0595ad97582f77b3282b9386b

dvesh3 committed:
Fix more merge conflicts in https://github.com/pimcore/pimcore/commit/504c00257e8ca9e0595ad97582f77b3282b9386b

dvesh3 committed:
[Pimcore 11] Improve version checks - follow up to #13896 (#13924)

Co-authored-by: Marco Perberschlager <marco.perberschlager@pimcore.com>

Corepex committed:
[Task]: Deprecated BC layer for `getO_` and `setO_` methods in dataobjects (#13886)

* deprecated bc layer for geto_ and seto_

* added upgrade notes

Commit:
enhance tag management: (#13844)

- resizable "assigned tags" panel when assigning tags to elements
- use full panel width for assigned element tags
- tag management: add context menu item to reload tags

fashxp committed:
Doc fixes (#13928)

* minor doc fixes

* minor doc fixes

SamyMP committed:
[Improvement]: Prompt a confirmation message when doing a batch modification (#13833)

* #13829 Prompt a confirmation message when doing a batch modification

* #13829 Prompt a confirmation message when doing a batch modification

fashxp committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

fashxp committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	models/DataObject/AbstractObject.php

fashxp committed:
minor doc fixes (#13929)

fashxp committed:
minor doc fixes (#13931)

jdreesen committed:
Fix cleanup via regexes in Service::getValidKey()

blankse committed:
AdminController: Use dependency injection (#13881)

dpfaffenbauer committed:
[Session] deprecate session class (#13923)

* [Session] deprecate session class

* Update lib/Tool/Session.php

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

* [Session] deprecate session class

* Update Session.php

* Update Session.php

* Update Session.php

* [Session] add upgrade notes

* clarify session variable

* Apply suggestions from code review

Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>

* Apply suggestions from code review

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>
Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>

brusch committed:
Apply php-cs-fixer changes

brusch committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	bundles/AdminBundle/Controller/ExtensionManager/ExtensionManagerController.php
#	lib/Tool/Session.php

brusch committed:
Fixed merge issue

kingjia90 committed:
Fix: key when document was quoted twice and small refactor (#13926)

mcop1 committed:
Apply php-cs-fixer changes

robertSt7 committed:
Fix: X-site-scripting (#13916)

dvesh3 committed:
[Assets] Fix text mime type mapping - resolves #13842

jdreesen committed:
Refactor Relation::getPhpDocClassString() (#13934)

mcop1 committed:
Apply php-cs-fixer changes

dvesh3 committed:
[Docs] Fix workflow management config - resolves #13750

robertSt7 commented:
@tecbird Have you had time to look at this?

mattamon committed:
12517 move glossary to bundle (#13811)

* Extract glossary from core

* Remove "use" of wrong installer

* Adding hook for keybinding registration
Adding migration for settings store
Moving DependencyInjection Configuration back to Core

* Remove permissions from installer

* Move translations to bundle

* Move GlossaryTest back to tests

* Making user keybindings configurable via yaml

* Reorder loading files in jsPaths and extract permissions to constant

* Add GlossaryBundle to ci/files/config/bundles.php

* Adding feedback changes
Adding warning to enable bundle manually
Removing unnecessary interfaces
Changing glossary.css to icons.css
Changing route names

* Adding feedback changes
Reverting routes
Adding bundle name to migration warning

* Adding sorted menu

* Removing unecessary commas

* Move migration to core/migrations

* Update Documentation, to enable and install the bundle

* Installing GlossaryBundle for testing

* Refactoring installation to own method.
Removing glossary installed check

* Moving translation back into core
Removing if condition from buildmenu

* Rename bundle from GlossaryBundle to PimcoreGlossaryBundle

* Adding Upgrade notes

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

* Adding extras menu by default and do not create if no items

* Adding method to key_bindings
Adding configuration for pimcore_glossary
Removing helper function

* Renaming method to action
Removing class dependency in migration

* Adding defaultkeybinding definition

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dvesh3 committed:
Apply php-cs-fixer changes

blankse committed:
Fix PhpStan

kingjia90 committed:
[Task] Remove `adminer` as Database Explorer (#13921)

* Task: remove adminer

* Update doc/Development_Documentation/23_Installation_and_Upgrade/03_System_Setup_and_Hosting/02_Nginx_Configuration.md

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* Update doc/Development_Documentation/23_Installation_and_Upgrade/03_System_Setup_and_Hosting/02_Nginx_Configuration.md

Co-authored-by: Sebastian Blank <blank@data-factory.net>

Co-authored-by: Sebastian Blank <blank@data-factory.net>

brusch committed:
Fixing PhpStan - cherry-picked from https://github.com/pimcore/pimcore/pull/13957 credits: @blankse

mcop1 committed:
[Bug]: QuantityValue Field returns '0' instead of 'NULL' after saving an empty Value (#13893)

dvesh3 committed:
[DataObject\Document] getChildren & getSiblings shouldn't load the listing (#13899)

* [DataObject\Document] getChildren & getSiblings shouldn't load the listing - resolves #11737

* [DataObject\Document] getChildren & getSiblings shouldn't load the listing - resolves #11737

* [DataObject\Document] getChildren & getSiblings shouldn't load the listing - resolves #11737

* [DataObject\Document] getChildren & getSiblings shouldn't load the listing - resolves #11737

robertSt7 committed:
Apply php-cs-fixer changes

blankse committed:
[Bug]: versions 500 error when field collection is removed (#13957)

* [Bug]: versions 500 error when field collection is removed

* Fix PhpStan

kingjia90 committed:
[Improvement]: Change default behavior of fallback documents (#13919)

* Task: remove of allow_processing_unpublished_fallback_document option and related deprecation triggers, set it as default

* Task: add upgrade note

* Task: remove ElementListener::FORCE_ALLOW_PROCESSING_UNPUBLISHED_ELEMENTS and its usage

* Task: improve changelog description of the change

mattamon committed:
Apply php-cs-fixer changes

robertSt7 committed:
Fix: Version in md-files (#13964)

* Fix: Version in CONTRIBUTING.md

* Fix: Version in some files

jdreesen committed:
Cleanup misplaced annotations (#13968)

dvesh3 committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

# Conflicts:
#	composer.json

mattamon committed:
Deprecate Parallelization::configureParallelization & using p shortcut on parallelized commands (#13938)

* Adding deprecation message and comment in console output if shortcut p is used with another option than processes

* Using server args to check for the -p shortcut

* Adding configureCommand for compatibility

* Update lib/Console/Traits/Parallelization.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Add upgrade notes

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dvesh3 committed:
Apply php-cs-fixer changes

kingjia90 committed:
Set `login_screen_custom_image` default to empty string to fix `twig_matches` error (#13948)

kingjia90 committed:
Set `login_screen_custom_image` default to empty string to fix `twig_matches` error (#13948)

Cruiser13 committed:
[Bug]: Avoid error of protected property starting with "\0" when removing reference loops (#13901)

Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>

brusch committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	lib/Tool/Text/Csv.php

fashxp committed:
[Installer] Fix eventually occurring 'too many connections' problem (#13962)

* close db connections during install

* close db connections during install

* close db connections during install

kingjia90 committed:
Revert "[Bug]: Avoid error of protected property starting with "\0" when removing reference loops (#13901)" (#13970)

This reverts commit ef9ccf5c9051b634ad796e6eedfd669eb61c70af.

kingjia90 committed:
[Bug]: Avoid error of protected property starting with "\0" when removing reference loops (#13971)

Co-authored-by: Lennart Fries <mail@lennartsauter.de>
Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>

mattamon committed:
13724 console parallelization (#13790)

* Adding options (main-process, segment-size)

* Added input options and removed deprecated methods
Added a deprecation warning for configureParallelization

* Add bundle name to deprecation note

* Removing configureParallelization method.
Using configureCommand Webmozarts\Console\Parallelization\Input\ParallelizationInput configureCommand directly
Breaking change since the p shortcut is not available anymore

* Adding upgrade notes

* Keep methods static
Removing extending options

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Changed configureCommand to static call
Updated upgrade notes

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dvesh3 committed:
Apply php-cs-fixer changes

dvesh3 committed:
[DataObject] Fix getDiffDataFromEditmode for multiselect  (#13965)

* [DataObject] Fix getDiffDataFromEditmode for multiselect - resolves https://github.com/pimcore/object-merger/issues/24

* [DataObject] Fix getDiffDataFromEditmode for multiselect - resolves https://github.com/pimcore/object-merger/issues/24

* Update models/DataObject/ClassDefinition/Data/Multiselect.php

Co-authored-by: Matthias S. <38959016+mattamon@users.noreply.github.com>

Co-authored-by: Matthias S. <38959016+mattamon@users.noreply.github.com>

jlsdohmen committed:
Fix: allow null as valid numeric filter condition (#13977)

Co-authored-by: Jules Dohmen <jdohmen@emico.nl>

mattamon committed:
[Task] Changing js class namespace, routing prefix and routing names (#13959)

* Changing js class namespace, routing prefix and routing names for consistency

* Changing routing config name

* Use namespace pimcore.bundle.glossary for all js files

* Moving translations to bundle to admin.en.json

* Change bundle_glossary to pimcoreBundleGlossary

torqdev committed:
Bug fix - allow export to contain duplicate columns (#13889)

* allow same name column on export

* use loop index instead

* remove unused variable

Co-authored-by: Cameron Jenkins <cameron.jenkins@torqit.ca>

robertSt7 committed:
Apply php-cs-fixer changes

kingjia90 committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

kingjia90 committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectHelperController.php
#	models/DataObject/ClassDefinition/Data/Multiselect.php

dvesh3 committed:
[Pimcore 11][DataObject] Fix getDiffDataFromEditmode for multiselect (#13966)

* [Pimcore 11][DataObject] Fix getDiffDataFromEditmode for multiselect #13965

* Update models/DataObject/ClassDefinition/Data/Multiselect.php

Co-authored-by: Matthias S. <38959016+mattamon@users.noreply.github.com>

* [Pimcore 11][DataObject] Fix getDiffDataFromEditmode for multiselect - revert duplicate

Co-authored-by: Matthias S. <38959016+mattamon@users.noreply.github.com>

robertSt7 committed:
Javascript: tag internal classes (#13954)

* Improve: Add internal tags in JS

* Improve: Add private tags in JS

* Apply suggestions from code review

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

robertSt7 committed:
[Bug]: Deleting a classificationstore group doesn't delete assigned data from an object (#13911)

* fix problem in grid configuration with block

An invisible Block in class breaks the grid configuration for folder grid

* Fix: Deleting a classificationstore group doesn't delete assigned data from an object

* Fix: Implemented reviewer feedback

* Update bundles/CoreBundle/Migrations/Version20221222181745.php

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* Update bundles/CoreBundle/Migrations/Version20221222181745.php

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* Fix: Deleting a classificationstore group doesn't delete assigned data from an object

Co-authored-by: frithjof <frithjof.pfannstiel@gfe-media.de>
Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

mcop1 committed:
Apply php-cs-fixer changes

Commit:
fix deadlock while saving assets concurrently

brusch committed:
[Actions] POEditor export: better config and added reference information on API call

brusch committed:
[Actions] POEditor export: better config and added reference information on API call - removed debug code

brusch committed:
[Actions] POEditor export: added glossary bundle translations

brusch committed:
[Actions] POEditor export: test

brusch committed:
[Actions] POEditor export: test

brusch committed:
[Actions] POEditor export: test

brusch committed:
[Actions] POEditor export: use add terms instead if sync_terms

brusch committed:
[Actions] POEditor export: test

mattamon committed:
[Task]: Move SEO functionalities to SeoBundle (#13851)

* Extract seo from admin
* Readd seo menu via bundle
* Adding new routes, installer and migration and templates
* Adding translations
Removing keybinding helpers (will be reintroduced with new event)
* Add SeoBundle to ci/files/config/bundles.php
* Adding feedback changes
Adding warning to enable bundle manually
Removing unnecessary interfaces
Changing route names
* Adding feedback changes
Reverting routes
Adding bundle name to migration warning
* Renaming SeoBundle to PimcoreSeoBundle
* Moving translations back into core
* Added upgrade notes
Added menu items
Added key bindings
* Changing js class namespace, routing prefix and routing names for consistency
Moving http_error_log table to bundle and installer
Adding Exceptionlistener with higher priority
* Rename pimcoreReady function to preMenuBuild
* Remove unecessary import
* Refactor JS namespaces and extracting translations to admin.en.json
* Javascript: tag private classes
* Making constant DOCUMENT_ROOT_ID privat
* Remove additonal seo string

kingjia90 committed:
Apply php-cs-fixer changes

mcop1 committed:
[Task]: Revert Bug fix - allow export to contain duplicate columns (#14010)

brusch committed:
[Actions] Removed POEditor export/import action -> moved to it's own repo + added trigger for new repo

blankse committed:
DynamicTextLabelInterface: params type should be an array (#13951)

* DynamicTextLabelInterface: params type should be an array

* Add params php doc

* $context param should also be array<string, mixed>

brusch committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	.github/ci/scripts/poeditor-export.php
#	.github/ci/scripts/poeditor-import.php
#	.github/workflows/poeditor-export.yaml
#	.github/workflows/poeditor-import.yaml
#	bundles/CoreBundle/src/Migrations/Version20221222181745.php
#	bundles/CoreBundle/translations/en.json
#	models/DataObject/Classificationstore/Dao.php

brusch committed:
en.json merge fix

brusch committed:
Merge remote-tracking branch 'origin/11.x' into 11.x

brusch committed:
classificationstore merge fix

kingjia90 committed:
Task: followup BuildMenu, remove empty items

blankse committed:
Cast request params to right type (#13981)

* Cast request params to right type

* Fields param can be null

* Cast request params to right type

* Optimize addCollectionsAction

* Use getBoolean to support false value (GET params are strings and "false" string is true)

* Cast request params to right type

* Default empty string for trim

* Cast request params to right type

* Cast request params to right type

* Cast request params to right type

* Prefer getInt() instead (int) casting

mattamon committed:
Add down method to migration

jdreesen committed:
Modernize element type handling (#13997)

* Modernize element type handling
* Add upgrade note
* Removed `Service::getType()`, use `Service::getElementType()` instead

blankse committed:
Fix data object csv export (#13989)

* Fix data object csv export

* Fix operator columns (for example char counter)

robertSt7 committed:
[Improvement]: JS Events cancelable (#13955)

* Improve: Make pre-events cancelable

* Improve: Make pre-events cancelable

blankse committed:
AdminController: Remove deprecated getters (#13950)

* AdminController: Remove deprecated getters
* AdminController: Add missing return types

brusch committed:
[Translator] Use YAML file format and default Symfony loading

brusch committed:
[Translator] Use YAML file format and default Symfony loading - follow up to #14027

brusch committed:
Apply php-cs-fixer changes

brusch committed:
fixed paths in POEditor action

brusch committed:
translation export trigger test

brusch committed:
Revert "translation export trigger test"

This reverts commit 8fe49a164ba12313d5e7c2ce6abd4be9cfed16f8.

mattamon committed:
Apply php-cs-fixer changes

aashan10 committed:
Made method results consistent between CartItemInterface and CheckoutableInterface (#13946)

* Made method results consistent between CartItemInterface and CheckoutableInterface

* Fixed conflicting type

* Fixed missing types

* Fixed phpstan issues

* Added default currency and validated prices

* Fixed mismatching type

* Made PriceSystem & AvailabilitySystem methods non-nullable in CheckoutableInterface - related to #9405

* Add upgrade notes

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>
Co-authored-by: Christian Fasching <fashxp@users.noreply.github.com>

mcop1 committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

fashxp committed:
[DataObjects] Consistent way of storing null/empty textfields to the database (#13963)

* first draft for allowing null values stored to db

* fix some setters

* fix some setters

* Revert "fix some setters"

This reverts commit 6fd56bb059f14359a9360abb31027102f27fd216.

* Revert "fix some setters"

This reverts commit 07cd8436a0dc0651fcce4505ce8be2a18d30c3b3.

* implement different approach for set empty values when loading from DB

* implement different approach for set empty values when loading from DB

* implement different approach for set empty values when loading from DB

* [DataObjects] Consistent way of storing null/empty textfields to the database
Checked Text, Number and date types

* Upgrade notes and tests

Co-authored-by: Marco Perberschlager <marco.perberschlager@pimcore.com>

fashxp committed:
Apply php-cs-fixer changes

mcop1 committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

mcop1 committed:
Merge remote-tracking branch 'origin/11.x' into 11.x

dpfaffenbauer committed:
NGINX Config - Add info about remote storages (S3, GCS) (#13840)

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

mcop1 committed:
Fix for phpstan issues (#14037)

mcop1 committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

mcop1 committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

mcop1 committed:
[Task]: Fix phpstan issues (#14034)

* Changed type for encodeFunc parameter

* Update bundles/AdminBundle/src/Controller/Admin/Asset/AssetHelperController.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* Fix phpstan issues

* Fix phpstan issues

* Fix phpstan issues

* Fix phpstan issues

Co-authored-by: Sebastian Blank <blank@data-factory.net>

Commit:
[Assets] Support document (PDF) version preview (#13858)

* support document (PDF) version preview
* add version date as download file name suffix
* use ExtJS to format date to have same display logic as in versions panel

Commit:
[Data Objects] Add select mode to many-to-one relation field (#13676)

* add display mode to many-to-one relation field

* add display mode to many-to-one relation field

* fix PhpStan error

* better describing option labels for "display mode" configuration

* simplify type / subtype / class filtering

* simplify type / subtype / class filtering

* simplify code

* refactoring

* work with Data object instead of its settings as array

* work with Data object instead of its settings as array

* add fullpath to visible fields

* add fullpath to visible fields

* fix PhpStan errors

* fix PhpStan errors - use Relation trait already in AbstractRelations as all Relation field types use it anyway, either directly or via inheritance

* add getVisibleFields() to many-to-one-relation. currently hard-coded to "fullpath". soon we will implement configuration of visible fields also for many-to-one relation fields

* support path formatter for select display mode for many-to-one-relation fields

* fix path formatter usage

* fix set value from search popup

* respect path formatter for select field mode

* Remove interfaces which already get provided by parent class

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* fix PhpStan

* fix PhpStan

* fix PhpStan

* fix PhpStan

* change translations to YAML

* fix PhpStan

* fix PhpStan

* fix PhpStan

* fix PhpStan

* fix PhpStan

* fix PhpStan

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

mcop1 committed:
Apply php-cs-fixer changes

Commit:
[Application logger] Support loading big FileObject files (#13995)

* support loading big FileObject files

* extract method to prevent duplicate code

lukmzig committed:
Apply php-cs-fixer changes

dvesh3 committed:
[ListCommand] Add details option for list commmand (#13961)

* [ListCommand] Add details option for list commmand - follow up to #13393

* Update bundles/CoreBundle/src/Command/Bundle/ListCommand.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

Co-authored-by: Sebastian Blank <blank@data-factory.net>

woellundwoell committed:
[issue-13984] Fix error message for duplicate URL slug (#13987)

Co-authored-by: Marc Weinreich <marc.weinreich@woellundwoell.de>
Co-authored-by: JiaJia Ji <kingjia90@gmail.com>

NiklasBr committed:
Sort Predefined Document Types (#12708)

* Sort Predefined Document Types

By priority first and name second

* Make sure document_types are sorted the same way as Priority is shown

* Improved sorting of DocTypes Listing

* Changed input type of Priority to NumberField

* Adjusted config for NumberField
- Slightly smaller
- Make editable (allows keyboard nav)
- Allow empty values

* Refactored Listing to move default sorting into Dao and make it smarter

* Update models/Document/DocType/Listing/Dao.php

Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>

* Remove accidental extra bracket from previous commit

* Remove unnecessary casts to int in DocType.php

* Update NumberField to remove allowBlank and set minValue to 1 (which was the previous min)

* Update bundles/AdminBundle/public/js/pimcore/settings/docTypes.js

Co-authored-by: Matthias S. <38959016+mattamon@users.noreply.github.com>

Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>
Co-authored-by: Matthias S. <38959016+mattamon@users.noreply.github.com>

robertSt7 committed:
Fix: Pimcore X by default displays deprecations directly in output (even in prod mode), breaking JSON etc (#13940)

fashxp committed:
Merge remote-tracking branch 'origin/10.5' into 10.6

fashxp committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	models/DataObject/ClassDefinition/Data/UrlSlug.php

fashxp committed:
Apply php-cs-fixer changes

fashxp committed:
Fixed phpdoc types -  (#14051)

* fixed phpdoc types - in preparation for #13869

* Update bundles/EcommerceFrameworkBundle/CartManager/CartPriceCalculatorInterface.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* Update bundles/EcommerceFrameworkBundle/IndexService/Getter/TagsGetter.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* Apply suggestions from code review

Co-authored-by: Sebastian Blank <blank@data-factory.net>

* fixed phpdoc types - in preparation for #13869

* fixed phpdoc types - in preparation for #13869

* fixed phpdoc types - in preparation for #13869

* Apply suggestions from code review

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Sebastian Blank <blank@data-factory.net>

Co-authored-by: Sebastian Blank <blank@data-factory.net>
Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

fashxp committed:
Apply php-cs-fixer changes

fashxp committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	bundles/EcommerceFrameworkBundle/src/CartManager/CartInterface.php
#	bundles/EcommerceFrameworkBundle/src/CartManager/CartPriceCalculatorInterface.php
#	bundles/EcommerceFrameworkBundle/src/CartManager/SessionCartCheckoutData.php
#	bundles/EcommerceFrameworkBundle/src/CoreExtensions/ClassDefinition/IndexFieldSelection.php
#	bundles/EcommerceFrameworkBundle/src/CoreExtensions/ClassDefinition/IndexFieldSelectionCombo.php
#	bundles/EcommerceFrameworkBundle/src/CoreExtensions/ClassDefinition/IndexFieldSelectionField.php
#	bundles/EcommerceFrameworkBundle/src/DependencyInjection/ServiceLocator/AssortmentTenantAwareServiceLocator.php
#	bundles/EcommerceFrameworkBundle/src/DependencyInjection/ServiceLocator/CheckoutTenantAwareServiceLocator.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Getter/DefaultBrickGetterSequenceToMultiselect.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Getter/TagsGetter.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/AssetId.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/DefaultObjects.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/DefaultRelations.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/DefaultStructuredTable.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/IdList.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/InterpreterInterface.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/Numeric.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/ObjectId.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/ObjectIdSum.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/Soundex.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Interpreter/StructuredTable.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/ProductList/DefaultMysql.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/ProductList/DefaultMysql/Dao.php
#	bundles/EcommerceFrameworkBundle/src/IndexService/Worker/Helper/MySql.php
#	bundles/EcommerceFrameworkBundle/src/Model/AbstractProduct.php
#	bundles/EcommerceFrameworkBundle/src/Model/CheckoutableInterface.php
#	bundles/EcommerceFrameworkBundle/src/OfferTool/AbstractOfferToolProduct.php
#	bundles/EcommerceFrameworkBundle/src/OfferTool/DefaultService.php
#	bundles/EcommerceFrameworkBundle/src/PaymentManager/V7/Payment/RecurringPaymentInterface.php
#	bundles/EcommerceFrameworkBundle/src/PriceSystem/AbstractPriceInfo.php
#	bundles/EcommerceFrameworkBundle/src/PriceSystem/LazyLoadingPriceInfo.php
#	bundles/EcommerceFrameworkBundle/src/PricingManager/Condition/VoucherToken.php
#	bundles/EcommerceFrameworkBundle/src/VoucherService/DefaultService.php
#	bundles/EcommerceFrameworkBundle/src/VoucherService/Token.php
#	bundles/EcommerceFrameworkBundle/src/VoucherService/Token/Listing.php
#	bundles/EcommerceFrameworkBundle/src/VoucherService/TokenManager/Pattern.php
#	bundles/EcommerceFrameworkBundle/src/VoucherService/TokenManager/Single.php
#	bundles/EcommerceFrameworkBundle/src/VoucherService/VoucherServiceInterface.php
#	lib/Db/Helper.php
#	lib/Image/Adapter/Imagick.php
#	lib/Localization/IntlFormatter.php
#	lib/Model/Dao/DaoInterface.php
#	lib/Workflow/SupportStrategy/ExpressionSupportStrategy.php
#	models/DataObject/ClassDefinition/Data/BooleanSelect.php
#	models/Document/Hardlink/Wrapper.php
#	models/Document/Hardlink/Wrapper/WrapperInterface.php
#	phpstan-baseline.neon

fashxp committed:
Apply php-cs-fixer changes

blankse committed:
Bump PhpStan Level to 6: Add missing param/return types (#13869)

* Add missing param/return types

* setValuesToDocument: Rename $page param to $document

* Add missing param/return types

* Fix PhpStan errors

* Add missing param/return types

* Update auf PhpStan 1.9.5

* Add missing param/return types

* Fix PhpStan

* Fix PhpStan

* Fix PhpStan

* Fix PhpStan

* Add missing return types

* Add missing param/return types

* Review

* Fix wrong return types

* Use return type self

* Use return type static

* Fix PhpStan

* Remove changes from tests (separate PR)

* Bump PhpStan Level to 6 with checkGenericClassInNonGenericObjectType/checkMissingIterableValueType = false

* Fix Codeception

* Bump Twig version to ^3.3.9 to have PhpDocs for SecurityPolicyInterface

* Add missing param/return types

* Add missing param/return types

* make params optional

* Pimcore\Tool\Console: Fix types

* Add missing return types

Co-authored-by: Christian Fasching <fashxp@users.noreply.github.com>

fashxp committed:
Apply php-cs-fixer changes

aashan10 committed:
Task: Extracted out System Info bundle (#13887)

* Extracted out System Info bundle
* Moved css to System info bundle
* Fixed bundle class names and integrated new menu system
* Removed bc layer for the extras menu
* Removed overridden getExtension method from bundle
* Update bundles/SystemInfoBundle/src/Controller/SettingsController.php
* move pimcore:system:requirements command
* Add Docs & Upgrade Notes
* Core tests should not depend on system info command
* new translation changes
* Updated types

Co-authored-by: Sebastian Blank <sebastian.bl@gmx.de>
Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>
Co-authored-by: JiaJia Ji <kingjia90@gmail.com>

kingjia90 committed:
Apply php-cs-fixer changes

brusch committed:
POEditor translations import (#14061)

Co-authored-by: Translations Bot <devs@pimcore.com>

dvesh3 committed:
[AssetUpdateTask] Route AssetUpdateTasks messages via dedicated transport - resolves #11613 (#13982)

Co-authored-by: aryaantony92 <97134765+aryaantony92@users.noreply.github.com>

brusch committed:
POEditor translations import (#14063)

Co-authored-by: Translations Bot <devs@pimcore.com>

aashan10 committed:
Extracted uuid to bundle (#13945)

* Extracted uuid to bundle

* Refactored UUID bundle according to the guidelines

* Move EventListener from Core to Bundle

* Update Upgrade notes

* Fixing nullsafe call on non-nullable type

* Fix if condition is always true

* Enabled UUID bundle in github actions

* Removed repetitive configs

* Fixed phpstan issues

* Update bundles/UuidBundle/src/PimcoreUuidBundle.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Fixed phpstan issues

* Updated the docs and dano classmaps

* Added migration to enable bundle

* Updated uuid bundle to use new configuration

Co-authored-by: mattamon <matthias.schuhmayer@pimcore.com>
Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dvesh3 committed:
Apply php-cs-fixer changes

mcop1 committed:
[Task]: Deprecate event "pimcore.element.note.postAdd" (#13835)

* [Task]: Deprecate event "pimcore.element.note.postAdd"

* [Task]: Deprecate event "pimcore.element.note.postAdd"
Added deprecation message + upgrade notes

Corepex committed:
Apply php-cs-fixer changes

Corepex committed:
[Docs] Improved perspective and custom view documentation by yaml code examples (#13915)

* improved documentation by converting php examples to yaml

* fixed typo

* added a link to the perspective editor

* updated documentation

* updated documentation

* Update doc/Development_Documentation/18_Tools_and_Features/13_Perspectives.md

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

* Update doc/Development_Documentation/18_Tools_and_Features/13_Perspectives.md

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

* added example configuration page

* Update doc/Development_Documentation/18_Tools_and_Features/13_Perspectives.md

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* Update doc/Development_Documentation/18_Tools_and_Features/13_Perspectives.md

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* added a link to the perspective-editor bundle

* added config file section to docs

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>
Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

dvesh3 committed:
Throw exception in Concrete:getClass(), if class not found (#13795)

* Throw exception in Concrete:getClass(), if class not found

* Throw exception in Concrete:getClass(), if class not found

* Add upgrade notes

* added error logging

Co-authored-by: Corepex <christian.feldkirchner@pimcore.com>

Corepex committed:
Apply php-cs-fixer changes

SamyMP committed:
[Improvement]: Cancel batch processing in object grid (#13839)

* Cancel batch processing in object grid

* Rebase 11.x

blankse committed:
Pimcore\Tool\Console: Remove deprecated methods (#14024)

* Pimcore\Tool\Console: Remove deprecated methods

* Remove redundant php doc

* Apply suggestions from code review

mcop1 committed:
[Task]: Create getter method for major version constant (#14070)

* Create getter method for major version constant

* Update lib/Version.php

Co-authored-by: Sebastian Blank <blank@data-factory.net>

Co-authored-by: Sebastian Blank <blank@data-factory.net>

mcop1 committed:
Merge remote-tracking branch 'origin/10.6' into 11.x

# Conflicts:
#	bundles/AdminBundle/src/Controller/Admin/IndexController.php
#	lib/Event/NoteEvents.php
#	lib/Version.php
#	models/Element/Note.php

tecbird committed:
move static definition to abstract class

Signed-off-by: Felix Pfister <github@felix-pfister.de>

tecbird committed:
Merge remote-tracking branch 'origin/feature/grid_config_relation_max' into feature/grid_config_relation_max

tecbird committed:
Merge branch 'feature/grid_config_relation_max2' into feature/grid_config_relation_max

# Conflicts:
#	bundles/AdminBundle/public/js/pimcore/object/helpers/grid.js

tecbird commented:
hey robert, i had some problems with rebasing in this mr, so i opened a new "clean" one https://github.com/pimcore/pimcore/pull/14078
